src/main/java/com/google/devtools/build/lib/remote/Chunker.java

```diff
@@ -142,15 +142,11 @@ public void reset() throws IOException {
    * <p>Closes any open resources (file handles, ...).
    */
   public void seek(long toOffset) throws IOException {
-    maybeInitialize();
     if (toOffset < offset) {
       reset();
-      if (toOffset != 0) {
-        data.skip(toOffset);
-      }
-    } else if (offset != toOffset) {
-      data.skip(toOffset - offset);
     }
+    maybeInitialize();
+    data.skip(toOffset - offset);
```

Comment:
While you are at it: I just noticed another bug with this code. We ignore the return value of InputStream#skip. According to its documentation it's free to skip less bytes than specified. Similar to read.

Do you want to fix this here or should we do it in a different change?

Reply:
I'll do it here

Reply:
There are crazy directions to go in with the response from the skip here: more than expected, 0 with a possible non-eof. I'll try to accommodate them all here, and throw if we get into a bad state. lmk if it's too much

Reply:
Nevermind, I'll just switch to using ByteStreams.skipFully, it accomplishes all that I wanted it to.

Reply:
Today I learned. Beautiful. Thanks!